packages/core/src/AttesterSchema.ts

```diff
@@ -0,0 +1,278 @@
+            throw Error('Schema includes an invalid key')
+        }
+
+        if (entry['type'] == 'uint') {
+            entry['type'] = 'uint253'
+        } else if (entry['type'] == 'bool') {
+            entry['type'] = 'uint1'
+        }
+
+        if (
+            String(entry['type']).slice(0, 4) != 'uint' ||
+            isNaN(+String(entry['type']).slice(4))
+        ) {
+            throw Error('Schema includes an invalid type')
+        }
+
+        if (entry['updateBy'] != 'sum' && entry['updateBy'] != 'replace') {
+            throw Error('Schema includes an invalid update method')
+        }
+    })
+    return schema
+}
+
+export class AttesterSchema {
+    private sumDataFields: DataField[]
+    private replDataFields: DataField[]
+
+    private curSumDataFieldCount: number
+    private curReplDataFieldCount: number
+    private schema: {}[]
+    private schemaLookup: {}
+
+    private encodedData: bigint[]
+
+    constructor(_schema: {}[]) {
+        this.curSumDataFieldCount = 0
+        this.curReplDataFieldCount = 0
+
+        this.sumDataFields = new Array(SUM_FIELD_COUNT)
+        this.replDataFields = new Array(FIELD_COUNT - SUM_FIELD_COUNT)
+
+        for (let i = 0; i < this.sumDataFields.length; i++) {
+            this.sumDataFields[i] = new DataField()
+        }
+
+        for (let i = 0; i < this.replDataFields.length; i++) {
+            this.replDataFields[i] = new DataField()
+        }
+
+        this.schema = parseSchema(_schema)
+        this.encodedData = new Array(FIELD_COUNT).fill(BigInt(2) ** BigInt(254))
+
+        this.encodedData = new Array(FIELD_COUNT).fill(BigInt(0))
+
+        this.schemaLookup = {}
+
+        this.buildDataFields()
+    }
+
+    findEntryByName(entry: string): {} {
+        for (let i = 0; i < this.schema.length; i++) {
+            if (this.schema[i]['name'] == entry) {
+                return Object.assign({}, this.schema[i])
+            }
+        }
+        return {}
+    }
+
+    buildDataFields() {
+        for (let i = 0; i < this.schema.length; i++) {
+            const name = this.schema[i]['name']
+            const bits = BigInt(this.schema[i]['type'].slice(4))
+            const updateBy = this.schema[i]['updateBy']
+
+            if (updateBy == 'sum') {
+                if (
+                    !this.sumDataFields[this.curSumDataFieldCount].addOrPass(
+                        name,
+                        bits
+                    )
+                ) {
+                    this.curSumDataFieldCount++
+
+                    if (this.curSumDataFieldCount >= SUM_FIELD_COUNT) {
+                        throw Error('Too many sum fields')
+                    }
+                    const res = this.sumDataFields[
+                        this.curSumDataFieldCount
+                    ].addOrPass(name, bits)
+
+                    if (!res) {
+                        throw Error("Couldn't add " + name + ' sum data field')
```

Comment:
this could be tested

Reply:
Using the same logic of repeatedly looking for available field indices. This error will no longer be reached.